### ISSUE TITLE
Local development notes on how to run tests from within a docker container

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/systemdspawner#123]
I think it would be good to help users test this software from within a docker container with systemd running. Like that, users wouldn't need to have systemd installed for example, and we could let systemdspawner run tests with system users without associated risks of doing it on the host system directly.

I don't think its in scope for our test suite to automatically build a docker image to run tests in, as we then need to maintain those image builds etc. With that said, I think its in scope to have an example Dockerfile with notes on how to build it, start it with the local folder mounted, and run pytest from inside it.

For reference, the TLJH distribution of jupyterhub let's pytest build a systemd image and run tests inside it etc. I think we should avoid that complexity in this project, but I think we can copy paste a lot from [that project's Dockerfile that includes `systemd`](https://github.com/jupyterhub/the-littlest-jupyterhub/blob/main/integration-tests/Dockerfile).

**Comment by manics** [comment]
I agree we shouldn't maintain another systemd container image in this repo. However since the work has already been done in TLJH and we're maintaining it, could we move that systemd container into a separate repo, and use it in both projects?
